source/elements/oneDAL/source/algorithms/nearest_neighbors/knn_classification.rst

```diff
@@ -1,3 +1,198 @@
+
+:math:`k`-NN classification algorithm infers the class for the new feature
+vector by computing majority vote of the :math:`k` nearest observations from the
+training set.
+
+.. |t_math| replace:: `Training <t_math_>`_
+.. |t_api| replace:: `API <t_api_>`_
+.. |t_brute_f| replace:: `Brute-force <t_math_brute_force_>`_
+.. |t_kd_tree| replace:: `k-d tree <t_math_kd_tree_>`_
+.. |t_input| replace:: `train_input <t_api_input_>`_
+.. |t_result| replace:: `train_result <t_api_result_>`_
+.. |t_op| replace:: `train(...) <t_api_>`_
+
+.. |i_math| replace:: `Inference <i_math_>`_
+.. |i_api| replace:: `API <i_api_>`_
+.. |i_brute_f| replace:: `Brute-force <i_math_brute_force_>`_
+.. |i_kd_tree| replace:: `k-d tree <i_math_kd_tree_>`_
+.. |i_input| replace:: `infer_input <i_api_input_>`_
+.. |i_result| replace:: `infer_result <i_api_result_>`_
+.. |i_op| replace:: `infer(...) <i_api_>`_
+
+
+=============== ============= ============= ======== =========== ============
+ **Operation**  **Computational methods**     **Programming Interface**
+--------------- --------------------------- ---------------------------------
+   |t_math|      |t_brute_f|   |t_kd_tree|   |t_op|   |t_input|   |t_result|
+   |i_math|      |i_brute_f|   |i_kd_tree|   |i_op|   |i_input|   |i_result|
+=============== ============= ============= ======== =========== ============
+
+------------------------
+Mathematical formulation
+------------------------
+
+.. _t_math:
+
+Training
+--------
+Given the training set :math:`X = \{ x_1, \ldots, x_n \}` of
+:math:`p`-dimensional feature vectors, the vector of class labels :math:`y` and
+a positive integer :math:`k`, where :math:`y_i \in \{ 0, \ldots, c-1 \}`,
+:math:`1 \leq i \leq n`, :math:`c` is the number of classes and :math:`k` is
+number of nearest neighbors specified by the user. The problem is to build a
+model that allows efficient distance computation between the feature vectors in
+:math:`X` and :math:`X'` at the inference stage.
+
+
+.. _t_math_brute_force:
+
+Training method: *brute-force*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The training operation produces the model that simply stores all the
+feature vectors from the initial training set :math:`X`.
+
+
+.. _t_math_kd_tree:
+
+Training method: *k-d tree*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The training operation builds a :math:`k`-:math:`d` tree (for more details, see
+:ref:`kd_tree`) that partitions the training set :math:`X`.
+
+
+.. _i_math:
+
+Inference
+---------
+Given the set :math:`X' = \{ x_1', \ldots, x_m' \}` of :math:`p`-dimensional
+feature vectors, the user defined number of nearest neighbors :math:`k` and the
+model produced by the training operation. The inference operation predicts the
+label for each :math:`x_i'`, :math:`1 \leq i \leq n`, by performing the
+following steps:
+
+#. Identify the set :math:`N(x_i') \subseteq X` of the :math:`k` feature vectors
+   in the training set that are nearest to :math:`x_i'` with respect to the
+   Euclidean distance.
```

Comment:
it's not necessarily Euclidean as far as I know

Reply:
Currently, it supports only Euclidean distance, support for the other distances need to be discussed.